### PR TITLE
fix(jupyter-kernel): don't log errors from objects without a `Symbol.for("Jupyter.display")`

### PR DESCRIPTION
**Pull request by rgbkrk**
Fast follow up to #20537.

Before:

image

After:

image

### CODE REVIEW

File: cli/tools/jupyter/server.rs [hunk 1]
```diff
@@ -490,9 +490,9 @@ async fn get_jupyter_display(
     )
     .await?;
 
-  if let Some(exception_details) = &response.exception_details {
-    // TODO(rgbkrk): Return an error in userspace instead of Jupyter logs
-    eprintln!("Exception encountered: {}", exception_details.text);
+  if response.exception_details.is_some() {
```
Comment (mmastrac) [inline cli/tools/jupyter/server.rs:493]: I think what we should do instead is change the function above to something like so. This should allow us to determine the difference between an object with no display function and an error.

```
function (object) {
  const display = object[Symbol.for("Jupyter.display")]
  if (display) {
    return JSON.stringify(display());
  } else {
    return null;
  }
}
```

**Comment by mmastrac** [comment]
@rgbkrk Thanks for the follow up! 

I tested this below code out and I think it should be pretty robust against missing symbols, error conditions, etc. If you want to test and update your PR with it we can merge this:

```
async fn get_jupyter_display(
  session: &mut repl::ReplSession,
  evaluate_result: &cdp::RemoteObject,
) -> Result<Option<HashMap<String, serde_json::Value>>, AnyError> {
  let response = session
    .call_function_on_args(
      r#"function (object) {
        const display = object[Symbol.for("Jupyter.display")]
        if (display) {
          return JSON.stringify(display());
        } else {
          return null;
        }
      }"#
      .to_string(),
      &[evaluate_result.clone()],
    )
    .await?;

  if let Some(exception_details) = &response.exception_details {
    // If the object doesn't have a Jupyter.display method or it throws an
    // exception, we just ignore it and let the caller handle it.
    eprintln!("Exception encountered: {}", exception_details.text);
    return Ok(None);
  }

  match response.result.value {
    Some(serde_json::Value::String(json_str)) => {
      let Ok(data) = serde_json::from_str::<HashMap<String, serde_json::Value>>(&json_str) else {
        eprintln!(
          "Unexpected response from Jupyter.display: {json_str}"
        );
        return Ok(None);
      };

      if !data.is_empty() {
        return Ok(Some(data));
      }
    },
    Some(serde_json::Value::Null) => {
      return Ok(None);
    },
    _ => {
      eprintln!(
        "Unexpected response from Jupyter.display: {:?}",
        response.result
      )
    }
  }

  Ok(None)
}
```

File: cli/tools/jupyter/server.rs (hunk 1 above)
Comment (rgbkrk) [reply cli/tools/jupyter/server.rs:493]: Ok cool. I'm still getting a feel for how much we inject in js vs rust.

**Review by mmastrac — APPROVED** [review]
LGTM

**Pull request merged**